### PR TITLE
#175494032 making code coverage

### PR DESCRIPTION
**Pull request by EngMutabazi**
### What does this PR do?

- integrate coverall code coverage service

### Description of Task to be completed?

- this will show the coverage of your test codes.

### How should this be manually tested?

- first  go into your text editor and clone the repo:`git clone https://github.com/Soma-Technologies-Inc/visitAfrica-backend.git`

- restart the server by typing: npm run startDev

- test the app by typing: npm run test

- to show the coverage of your code test type: npm run coverage

### Any background context you want to provide?
N/A
### What are the relevant pivotal tracker stories?
[#175494032](https://www.pivotaltracker.com/story/show/175494032)
Screenshot (if appropriate)
Screenshot from 2020-11-21 19-50-26

### CODE REVIEW

File: src/helpers/helpers.test.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const chai=require('chai');
+const expect=chai.expect;
+const user=require('./index')
```
Comment (erickyvand) [inline src/helpers/helpers.test.js]: You have to use `ES6 syntax`

File: src/helpers/helpers.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+describe('User test cases:',()=>{
+
+it('should add a new user',()=>{
+user.addUser();
+
+console.log('added a user');
```
Comment (erickyvand) [inline src/helpers/helpers.test.js]: Why this `console.log`?

File: src/helpers/helpers.test.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+user.removeUser();
+console.log('removed a user');
```
Comment (erickyvand) [inline src/helpers/helpers.test.js]: Why this `console.log`?

File: src/helpers/index.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+module.exports={
```
Comment (erickyvand) [inline src/helpers/index.js]: use `ES6 syntax`

File: src/helpers/index.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+    addUser:function(){
+        console.log("new user added")
+    },
+    removeUser:function(){
+        console.log("new user removed");
+    }
+    
+    }
```
Comment (erickyvand) [inline src/helpers/index.js]: Add a new line at the end

**Review by erickyvand — CHANGES_REQUESTED** [review]

**Comment by erickyvand** [comment]
@engmutabazi100 On your description title you should add pivotal tracker story ID at the beginning. And also on `What are the relevant pivotal tracker stories` you should pivotal tracker ID to be clickable by using this format [put pivotal tracker story ID here](put pivotal tracker story link here)

**Comment by Niyonsengaeric** [comment]
@engmutabazi100  write a good and clear PR description!

File: coverage/index.html [hunk 6]
```diff
@@ -0,0 +1,111 @@
+
```
Comment (erickyvand) [inline coverage/index.html]: the `coverage` folder should be added in `.gitignore` file

File: .coveralls.yml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+
+service_name: travis-pro
+repo_token: nUINoLdenxYURJlssXO3pDfWXAY1n6Xbp
```
Comment (erickyvand) [inline .coveralls.yml:2]: put the value of that token in `.env` file then you can call the value in this file like this `${process.env.REPO_TOKEN}`. Also remember to add an example in `.env.example` of every variable you put in `.env`

File: coverage/index.html (hunk 6 above)
Comment (erickyvand) [reply coverage/index.html]: Means these codes should not be uploaded to github

File: src/server.js [hunk 8]
```diff
@@ -18,3 +18,5 @@ app.use('/',(req,res)=>res.status(404).json({status:404,
 
 const PORT = process.env.PORT || 3000
 app.listen(PORT, () =>console.log(`Sever listening on port ${PORT}`));
+
+export default app;
```
Comment (erickyvand) [inline src/server.js]: add new line

File: package.json [hunk 9]
```diff
@@ -38,10 +41,14 @@
...
   },
   "dependencies": {
-    "@babel/preset-stage-2": "^7.8.3",
+    "chai-http": "^4.3.0",
     "cors": "^2.8.5",
+    "cross-env": "^7.0.2",
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
-    "form-data": "^3.0.0"
+    "form-data": "^3.0.0",
```
Comment (erickyvand) [inline package.json]: why this package?

File: package.json [hunk 10]
```diff
@@ -38,10 +41,14 @@
     "nodemon": "^2.0.6"
   },
   "dependencies": {
-    "@babel/preset-stage-2": "^7.8.3",
+    "chai-http": "^4.3.0",
     "cors": "^2.8.5",
+    "cross-env": "^7.0.2",
```
Comment (erickyvand) [inline package.json]: why this package?

File: src/server.js (hunk 8 above)
Comment (erickyvand) [inline src/server.js]: add new line

File: package.json (hunk 10 above)
Comment (erickyvand) [inline package.json]: why this package?

File: package.json (hunk 9 above)
Comment (erickyvand) [inline package.json]: why this package?

**Review by erickyvand — CHANGES_REQUESTED** [review]

File: package.json [hunk 11]
```diff
@@ -23,8 +24,15 @@
   },
   "homepage": "https://github.com/Soma-Technologies-Inc/visitAfrica-backend#readme",
   "devDependencies": {
```
Comment (Niyonsengaeric) [inline package.json:26]: why babel in devDependencies?

**Review by Niyonsengaeric — CHANGES_REQUESTED** [review]

File: src/helpers/helpers.test.js (hunk 1 above)
Comment (EngMutabazi) [reply src/helpers/helpers.test.js]: solved

File: src/helpers/helpers.test.js (hunk 2 above)
Comment (EngMutabazi) [reply src/helpers/helpers.test.js]: solved

File: src/helpers/helpers.test.js (hunk 3 above)
Comment (EngMutabazi) [reply src/helpers/helpers.test.js]: solved

File: src/helpers/index.js (hunk 4 above)
Comment (EngMutabazi) [reply src/helpers/index.js]: babel was not configured before but now every thing works fine

File: src/helpers/index.js (hunk 5 above)
Comment (EngMutabazi) [reply src/helpers/index.js]: solved!

File: package.json (hunk 9 above)
Comment (EngMutabazi) [reply package.json]: these nothing to help but solved

File: package.json (hunk 11 above)
Comment (EngMutabazi) [reply package.json:26]: they should be in Dependencies but i solved it

File: package.json (hunk 10 above)
Comment (EngMutabazi) [reply package.json]: it is not needed and i solved it

File: src/server.js (hunk 8 above)
Comment (EngMutabazi) [reply src/server.js]: done now

File: package.json (hunk 10 above)
Comment (EngMutabazi) [reply package.json]: checked

File: package.json (hunk 9 above)
Comment (EngMutabazi) [reply package.json]: solved!

File: src/server.js (hunk 8 above)
Comment (EngMutabazi) [reply src/server.js]: done

File: coverage/index.html (hunk 6 above)
Comment (EngMutabazi) [reply coverage/index.html]: yes,solved

File: package.json (hunk 10 above)
Comment (Niyonsengaeric) [reply package.json]: @engmutabazi100  try to Read the use of this package you may need it

**Review by erickyvand — APPROVED** [review]

**Pull request merged**